### PR TITLE
Test restructure

### PR DESCRIPTION
**Pull request by FWDekker**
Restructures the tests as discussed IRL.

* `gradle/tests/test-configuration-generator.gradle` exports a function that allows you to generate new test configurations.
* The smoke tests from `:application` have been moved to `module-integration-tests`.
* `PrefixSpanJimpleNodeIntegrationTest` has been moved to `module-integration-tests`.

### CODE REVIEW

**Comment by FWDekker** [comment]
@gandreadis 
```
de
d

``` 
?

File: gradle/tests/test-configuration-generator.gradle [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            resources.srcDirs file("src/$directoryName/resources")
+
+            compileClasspath += sourceSets.main.output
+            compileClasspath += configurations.compileOnly
+            compileClasspath += configurations.testCompileOnly
+            runtimeClasspath += compileClasspath
+        }
+    }
+
+    def myTestTask = task(taskName, type: Test, group: "Verification") {
```
Comment (casperboone) [inline gradle/tests/test-configuration-generator.gradle]: Can we name this differently?

Comment (FWDekker) [reply gradle/tests/test-configuration-generator.gradle]: I can't think of a good alternative. Any suggestions?

Comment (gandreadis) [reply gradle/tests/test-configuration-generator.gradle]: `newTestTask`? `customTestTask`? `generatedTestTask`?

**Comment by FWDekker** [comment]
@casperboone Done

**Pull request merged**